internal/wasi/wasi_test.go

```diff
@@ -474,7 +474,35 @@ func TestAPI_ClockTimeGet_Errors(t *testing.T) {
 
 // TODO: TestAPI_FdAdvise TestAPI_FdAdvise_Errors
 // TODO: TestAPI_FdAllocate TestAPI_FdAllocate_Errors
-// TODO: TestAPI_FdClose TestAPI_FdClose_Errors
+
+func TestAPI_FdClose(t *testing.T) {
+	opt := Preopen("test", &MemFS{})
```

Comment:
Let's be more direct as it explains more what's going on. The only complicated API (ex that uses the filesystem) is fd_open. For the others we can be really direct as it make later assertions clear on side-effects.

```
fd := uint32(3) // example FD past stdin/out/err
otherFD := uint32(5)

var wasiAPI *wasiAPI
store, ctx, fn := instantiateWasmStore(t, FunctionFdClose, ImportFdClose, "test", func(a *wasiAPI) {
		a.opened = map[uint32]fileEntry{
                    fd: fileEntry{path: "to be closed"},
                    otherFD: fileEntry{path: "shouldn't be closed"},
                 }
wasiAPI = a // for later assertions
})
```